NumLib/ODESolver/NonlinearSolver-impl.h

```diff
@@ -160,18 +160,12 @@ solve(Vector &x)
         sys.assembleResidualNewton(x);
         sys.getResidual(x, res);
 
-        // std::cout << "  res:\n" << res << std::endl;
-
-        // TODO streamline that, make consistent with Picard.
-        if (BLAS::norm2(res) < _tol) {
-            error_norms_met = true;
-            break;
-        }
-
         sys.assembleJacobian(x);
         sys.getJacobian(J);
         sys.applyKnownSolutionsNewton(J, res, minus_delta_x);
 
+        auto const error_res = BLAS::norm2(res);
```

Comment:
Okay, if the code will be moved later we can keep it for the moment.

:fast_forward: